app/services/storage/s3/uploader.rb

```diff
@@ -4,18 +4,13 @@
 module Storage
   module S3
     class Uploader
-      def initialize(path:, key:, bucket:)
-        @path = Pathname.new(path)
+      def initialize(key:, bucket:)
         @key = key
         @bucket = bucket
       end
 
-      def upload
-        encrypt
-        File.open(path_to_encrypted_file, 'rb') do |file|
-          client.put_object(bucket: bucket, key: key, body: file)
-        end
-        File.delete(path_to_encrypted_file)
+      def upload(file: file)
+        client.put_object(bucket: bucket, key: key, body: file.read)
       end
```

Comment:
We could go one step further and upload a data blob rather than a `File`/`Tempfile`/`io` object. It would further reduce the need to write to disk but not sure about the implications of passing an encrypted data blob around.

Reply:
Implemented this